lib/rmt/lockfile.rb

```diff
@@ -4,13 +4,13 @@ class RMT::Lockfile
 
   class << self
     def create_file
-      if File.exist?(LOCKFILE_LOCATION)
-        raise RMT::Lockfile::ExecutionLockedError
-      else
-        dirname = File.dirname(LOCKFILE_LOCATION)
-        FileUtils.mkdir_p(dirname) unless File.directory?(dirname)
-        File.open(LOCKFILE_LOCATION, 'w') { |f| f.write(Process.pid) }
-      end
+      dirname = File.dirname(LOCKFILE_LOCATION)
+      FileUtils.mkdir_p(dirname) unless File.directory?(dirname)
+
+      # has to be a class instance variable otherwise garbage-collected and lock is removed
+      @f = File.open(RMT::Lockfile::LOCKFILE_LOCATION, File::RDWR | File::CREAT)
+      @f.write(Process.pid.to_s) if @f.read.empty? # write PID if file is new. this has to be done before locking
+      raise ExecutionLockedError unless @f.flock(File::LOCK_EX | File::LOCK_NB)
     end
 
     def remove_file
```

Comment:
Is removing the file enough? AFAIK, once you lock a file, you need to unlock it with `f.flock(File::LOCK_UN)`. See for example https://www.safaribooksonline.com/library/view/ruby-cookbook/0596523696/ch06s13.html

Reply:
we can either delete or unlock the file. as I checked it now - both options work